python/tests/test_combinatorics.py

```diff
@@ -481,6 +481,31 @@ def test_is_symmetrical(self):
         six_leaf_sym = RankTree(children=[three_leaf_asym, three_leaf_asym])
         assert six_leaf_sym.is_symmetrical()
 
+    @pytest.mark.parametrize("n", range(7))
+    def test_leaves(self, n):
+        for tree in RankTree.all_unlabelled_trees(n):
+            first_labelled_tree = RankTree.unrank(n, (tree.shape_rank(), 0))
+            assert list(first_labelled_tree.leaves()) == list(range(n))
+
+    @pytest.mark.parametrize("n", range(5))
+    def test_no_leaves(self, n):
+        for tree in RankTree.all_unlabelled_trees(n):
+            assert list(tree.leaves()) == [None] * n
+
+    @pytest.mark.parametrize("n", range(6))
+    def test_newick_order(self, n):
+        del_digits = {ord(c): "" for c in "1234567890"}
```

Comment:
Also seems excessive to do this for so many trees, since it's a fairly simple property which if it's true for a handful of examples will be true for everything.